tomviz/DataSource.cxx

```diff
@@ -123,6 +127,26 @@ void deserializeDataArray(const pugi::xml_node& ns, vtkDataArray* array)
 }
 }
 
+DataSource::ImageFuture::ImageFuture(Operator* op,
+                                     vtkSmartPointer<vtkImageData> imageData,
+                                     PipelineWorker::Future* future,
+                                     QObject* parent)
+  : QObject(parent), m_operator(op), m_imageData(imageData), m_future(future)
+{
+
+  if (m_future != nullptr) {
+    connect(m_future, SIGNAL(finished()), this, SIGNAL(finished()));
+    connect(m_future, SIGNAL(canceled()), this, SIGNAL(canceled()));
+  }
+}
+
+DataSource::ImageFuture::~ImageFuture()
+{
+  if (this->m_future != nullptr) {
```

Comment:
Minor style point that I'm not sure about: I don't think we are using `this->` in front of `m_var`. In any case, it's usage is inconsistent in this file.